Source/Scene/ModelExperimental/ModelExperimentalSceneGraph.js

```diff
@@ -0,0 +1,280 @@
+import Axis from "../../Scene/Axis.js";
+import buildDrawCommand from "./buildDrawCommand.js";
+import Check from "../../Core/Check.js";
+import defaultValue from "../../Core/defaultValue.js";
+import defined from "../../Core/defined.js";
+import Matrix4 from "../../Core/Matrix4.js";
+import ModelExperimentalSceneMeshPrimitive from "./ModelExperimentalSceneMeshPrimitive.js";
+import ModelExperimentalSceneNode from "./ModelExperimentalSceneNode.js";
+import ModelExperimentalUtility from "./ModelExperimentalUtility.js";
+import RenderResources from "./RenderResources.js";
+import BoundingSphere from "../../Core/BoundingSphere.js";
+
+/**
+ * An in memory representation of the scene graph for a {@link ModelExperimental}
+ *
+ * @param {Object} options An object containing the following options
+ * @param {ModelExperimental} options.model The model this scene graph belongs to
+ * @param {ModelComponents} options.modelComponents The model components describing the model
+ * @param {Axis} [options.upAxis=Axis.Y] The upwards direction of the 3D model
+ * @param {Axis} [options.forwardAxis=Axis.Z] The forwards direction of the 3D model
+ *
+ * @private
+ */
+export default function ModelExperimentalSceneGraph(options) {
```

Comment:
Is it too ambitious to hope for a uniform rendering API across all types of 3D geometry? Currently this only works for gltf (or tiles with gltf), but it'd be awesome if the scene graph and pipeline stages could live at a higher level..

Reply:
Not sure how easily that can be done, lots of other things would need to be refactored to use this architecture... @lilleyse were there any long-term plans like this? I'm not aware of any.

I will point out though that @lilleyse's intent with `ModelComponents` (the structs returned by `GltfLoader`) is to be more general, as long as you can build a ModelComponents you could potentially use this with our code (e.g. perhaps someday an ObjLoader or StlLoader or even just procedurally generating a mesh from typed arrays) with some tweaks to the constructor.